### ISSUE TITLE
Amplify backend build error

### ISSUE THREAD

**Issue by seunghwako** [aws-amplify/amplify-cli#10609]
### Before opening, please confirm:

- [X] I have checked to see if my question is addressed in the [FAQ](https://github.com/aws-amplify/amplify-hosting/blob/master/FAQ.md).
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-hosting/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-hosting/blob/master/CONTRIBUTING.md).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### App Id

d3u8gr5p05kbef

### Region

ap-northeast-2

### Amplify Hosting feature

Backend builds

### Describe the bug

Push the source code and build changes to the repository connected to the amplify. This error occurs in the backend build step and the build fails. I don't know why the error DeploymentBucketBlockHTTP occurs.

 ```
                           # Starting phase: build
2022-06-14T06:00:01.900Z [INFO]: [0mAmplify AppID found: d3u8gr5p05kbef. Amplify App name is: fine[0m
2022-06-14T06:00:01.970Z [INFO]: [0mBackend environment main found in Amplify Console app: fine[0m
2022-06-14T06:00:03.039Z [WARNING]: - Fetching updates to backend environment: main from the cloud.
2022-06-14T06:00:04.050Z [WARNING]: - Building resource auth/fined0f28391
2022-06-14T06:00:04.175Z [WARNING]: - Building resource auth/userPoolGroups
2022-06-14T06:00:04.287Z [WARNING]: - Building resource storage/s34284101e
2022-06-14T06:00:04.332Z [WARNING]: - Building resource api/fine
2022-06-14T06:00:06.291Z [INFO]: [32mGraphQL schema compiled successfully.[39m
                                 [32mEdit your schema at /codebuild/output/src844669505/src/fine/amplify/backend/api/fine/schema.graphql or place .graphql files in a directory at /codebuild/output/src844669505/src/fine/amplify/backend/api/fine/schema[39m
2022-06-14T06:00:06.294Z [WARNING]: - Building resource api/AdminQueries
2022-06-14T06:00:06.305Z [INFO]: Overrides functionality is not implemented for this category
2022-06-14T06:00:06.331Z [WARNING]: ✔ Successfully pulled backend environment main from the cloud.
2022-06-14T06:00:06.382Z [INFO]: ✅
2022-06-14T06:00:08.222Z [INFO]: [33mNote: It is recommended to run this command from the root of your app directory[39m
2022-06-14T06:00:09.136Z [WARNING]: - Initializing your environment: main
2022-06-14T06:00:10.109Z [WARNING]: - Building resource auth/fined0f28391
2022-06-14T06:00:10.233Z [WARNING]: - Building resource auth/userPoolGroups
2022-06-14T06:00:10.343Z [WARNING]: - Building resource storage/s34284101e
2022-06-14T06:00:10.389Z [WARNING]: - Building resource api/fine
2022-06-14T06:00:11.921Z [INFO]: [32mGraphQL schema compiled successfully.[39m
                                 [32mEdit your schema at /codebuild/output/src844669505/src/fine/amplify/backend/api/fine/schema.graphql or place .graphql files in a directory at /codebuild/output/src844669505/src/fine/amplify/backend/api/fine/schema[39m
2022-06-14T06:00:11.922Z [WARNING]: - Building resource api/AdminQueries
2022-06-14T06:00:11.935Z [INFO]: Overrides functionality is not implemented for this category
2022-06-14T06:00:11.935Z [WARNING]: ✔ Initialized provider successfully.
2022-06-14T06:00:17.704Z [WARNING]: - Uploading files...
2022-06-14T06:00:18.317Z [WARNING]: - Building resource auth/fined0f28391
2022-06-14T06:00:18.392Z [WARNING]: - Building resource auth/userPoolGroups
2022-06-14T06:00:18.459Z [WARNING]: - Building resource storage/s34284101e
2022-06-14T06:00:18.492Z [WARNING]: - Building resource api/fine
2022-06-14T06:00:19.939Z [INFO]: [32mGraphQL schema compiled successfully.[39m
                                 [32mEdit your schema at /codebuild/output/src844669505/src/fine/amplify/backend/api/fine/schema.graphql or place .graphql files in a directory at /codebuild/output/src844669505/src/fine/amplify/backend/api/fine/schema[39m
2022-06-14T06:00:19.940Z [WARNING]: - Building resource api/AdminQueries
2022-06-14T06:00:19.955Z [INFO]: Overrides functionality is not implemented for this category
2022-06-14T06:00:19.959Z [WARNING]: - Uploading files...
2022-06-14T06:00:20.874Z [WARNING]: - Updating resources in the cloud. This may take a few minutes...
2022-06-14T06:00:26.855Z [INFO]: 
2022-06-14T06:00:26.856Z [INFO]: UPDATE_IN_PROGRESS amplify-fine-main-181850  AWS::CloudFormation::Stack Tue Jun 14 2022 06:00:21 GMT+0000 (Coordinated Universal Time) User Initiated
                                 CREATE_IN_PROGRESS DeploymentBucketBlockHTTP AWS::S3::BucketPolicy      Tue Jun 14 2022 06:00:26 GMT+0000 (Coordinated Universal Time)
2022-06-14T06:00:31.894Z [INFO]: 
2022-06-14T06:00:31.895Z [INFO]: CREATE_FAILED               DeploymentBucketBlockHTTP AWS::S3::BucketPolicy      Tue Jun 14 2022 06:00:27 GMT+0000 (Coordinated Universal Time) The bucket policy already exists on bucket amplify-fine-main-181850-deployment.
                                 UPDATE_ROLLBACK_IN_PROGRESS amplify-fine-main-181850  AWS::CloudFormation::Stack Tue Jun 14 2022 06:00:28 GMT+0000 (Coordinated Universal Time) The following resource(s) failed to create: [DeploymentBucketBlockHTTP].
2022-06-14T06:00:36.861Z [INFO]: 
2022-06-14T06:00:36.862Z [INFO]: UPDATE_ROLLBACK_COMPLETE_CLEANUP_IN_PROGRESS amplify-fine-main-181850 AWS::CloudFormation::Stack Tue Jun 14 2022 06:00:36 GMT+0000 (Coordinated Universal Time)
2022-06-14T06:00:41.914Z [INFO]: 
2022-06-14T06:00:41.916Z [INFO]: DELETE_COMPLETE          DeploymentBucketBlockHTTP AWS::S3::BucketPolicy      Tue Jun 14 2022 06:00:37 GMT+0000 (Coordinated Universal Time)
                                 UPDATE_ROLLBACK_COMPLETE amplify-fine-main-181850  AWS::CloudFormation::Stack Tue Jun 14 2022 06:00:37 GMT+0000 (Coordinated Universal Time)
2022-06-14T06:00:51.880Z [INFO]: Following resources failed
                                 Resource Name: DeploymentBucketBlockHTTP (AWS::S3::BucketPolicy)
                                 Event Type: create
                                 Reason: The bucket policy already exists on bucket amplify-fine-main-181850-deployment.
2022-06-14T06:00:51.881Z [WARNING]: ✖ An error occurred when pushing the resources to the cloud
2022-06-14T06:00:51.881Z [WARNING]: ✖ There was an error initializing your environment.
2022-06-14T06:00:51.887Z [INFO]: [0mResourceNotReady: Resource is not in the state stackUpdateComplete[0m
                                 [0m    at constructor.setError (/snapshot/repo/build/node_modules/aws-sdk/lib/resource_waiter.js:182:47)[0m
                                 [0m    at Request.CHECK_ACCEPTORS (/snapshot/repo/build/node_modules/aws-sdk/lib/resource_waiter.js:44:12)[0m
                                 [0m    at Request.callListeners (/snapshot/repo/build/node_modules/aws-sdk/lib/sequential_executor.js:106:20)[0m
                                 [0m    at Request.emit (/snapshot/repo/build/node_modules/aws-sdk/lib/sequential_executor.js:78:10)[0m
                                 [0m    at Request.emit (/snapshot/repo/build/node_modules/aws-sdk/lib/request.js:686:14)[0m
                                 [0m    at Request.transition (/snapshot/repo/build/node_modules/aws-sdk/lib/request.js:22:10)[0m
                                 [0m    at AcceptorStateMachine.runTo (/snapshot/repo/build/node_modules/aws-sdk/lib/state_machine.js:14:12)[0m
                                 [0m    at /snapshot/repo/build/node_modules/aws-sdk/lib/state_machine.js:26:10[0m
                                 [0m    at Request.<anonymous> (/snapshot/repo/build/node_modules/aws-sdk/lib/request.js:38:9)[0m
                                 [0m    at Request.<anonymous> (/snapshot/repo/build/node_modules/aws-sdk/lib/request.js:688:12)[0m
                                 [0m    at Request.callListeners (/snapshot/repo/build/node_modules/aws-sdk/lib/sequential_executor.js:116:18)[0m
                                 [0m    at Request.emit (/snapshot/repo/build/node_modules/aws-sdk/lib/sequential_executor.js:78:10)[0m
                                 [0m    at Request.emit (/snapshot/repo/build/node_modules/aws-sdk/lib/request.js:686:14)[0m
                                 [0m    at Request.transition (/snapshot/repo/build/node_modules/aws-sdk/lib/request.js:22:10)[0m
                                 [0m    at AcceptorStateMachine.runTo (/snapshot/repo/build/node_modules/aws-sdk/lib/state_machine.js:14:12)[0m
                                 [0m    at /snapshot/repo/build/node_modules/aws-sdk/lib/state_machine.js:26:10[0m
                                 [0m    at Request.<anonymous> (/snapshot/repo/build/node_modules/aws-sdk/lib/request.js:38:9)[0m
                                 [0m    at Request.<anonymous> (/snapshot/repo/build/node_modules/aws-sdk/lib/request.js:688:12)[0m
                                 [0m    at Request.callListeners (/snapshot/repo/build/node_modules/aws-sdk/lib/sequential_executor.js:116:18)[0m
                                 [0m    at callNextListener (/snapshot/repo/build/node_modules/aws-sdk/lib/sequential_executor.js:96:12)[0m
                                 [0m    at IncomingMessage.onEnd (/snapshot/repo/build/node_modules/aws-sdk/lib/event_listeners.js:335:13)[0m
                                 [0m    at IncomingMessage.emit (events.js:412:35)[0m
                                 [0m    at IncomingMessage.emit (domain.js:475:12)[0m
                                 [0m    at endReadableNT (internal/streams/readable.js:1334:12)[0m
                                 [0m    at processTicksAndRejections (internal/process/task_queues.js:82:21) {[0m
                                 [0m  code: 'ResourceNotReady',[0m
                                 [0m  retryable: false,[0m
                                 [0m  time: 2022-06-14T06:00:51.877Z,[0m
                                 [0m  statusCode: 200,[0m
                                 [0m  retryDelay: 30000[0m
                                 [0m}[0m
2022-06-14T06:00:51.936Z [ERROR]: !!! Build failed
2022-06-14T06:00:51.937Z [ERROR]: !!! Non-Zero Exit Code detected
2022-06-14T06:00:51.937Z [INFO]: # Starting environment caching...
2022-06-14T06:00:51.937Z [INFO]: # Uploading environment cache artifact...
2022-06-14T06:00:52.019Z [INFO]: # Environment caching completed
Terminating logging...
```

### Expected behavior

The build step must be completed successfully and the deployment step must be moved on

### Reproduction steps

1. Change in vscode git push
2. Start provisioning in amplify
3. Start repository clone after provisioning is complete
4. Start backend build after repository clone is complete
5. Error occurred

### Build Settings

```yaml
version: 1
backend:
  phases:
    build:
      commands:
        - '# Execute Amplify CLI with the helper script'
        - amplifyPush --simple
frontend:
  phases:
    preBuild:
      commands:
        - export NODE_OPTIONS=--max-old-space-size=8192
        - yarn install
    build:
      commands:
        - yarn run build
  artifacts:
    baseDirectory: .next
    files:
      - '**/*'
  cache:
    paths:
      - node_modules/**/*
      - .next/cache/**/* # Cache Next.js for faster application rebuilds
```

**Comment by seunghwako** [comment]
@hloriana Thank you. If the problem is solved in the CLI team, can I share it?

**Comment by sachscode** [comment]
Hi , 
This problem was identified and resolved in amplify-cli 8.5.1 . In this particular case , the deployment bucket failed to create because the amplify generated cloudformation and AWS were both trying to apply the block-public-access policy on the bucket. This causes the cloudformation deployment to fail wither with the error:  "A conflicting conditional operation is currently in progress" Or "The bucket policy already exists on bucket" depending on whether on not AWS was able to apply the policy prior to the deployment. Upgrading to 8.5.1 and above will solve this problem.
